FlowCrypt/Controllers/Msg/MessageViewController.swift

```diff
@@ -163,16 +163,18 @@ extension MessageViewController {
 
     private func handlePassPhraseEntry(rawMimeData: Data, with passPhrase: String) {
         showSpinner("loading_title".localized, isUserInteractionEnabled: true)
-        do { // todo - should be a task on non-main thread?
-            let matched = try messageService.checkAndPotentiallySaveEnteredPassPhrase(passPhrase)
-            if matched {
-                self.processedMessage = try awaitPromise(self.messageService.decryptAndProcessMessage(mime: rawMimeData))
-                self.handleReceivedMessage()
-            } else {
-                handleWrongPathPhrase(for: rawMimeData, with: passPhrase)
+        Task {
```

Comment:
Since you don't have `async/await` inside Task, you don't need it

Reply:
Thanks! Updating.

Reply:
I'll change it back :-) `checkAndPotentiallySaveEnteredPassPhrase` is now async anyway.